optimism/src/keccak/constraints.rs

```diff
@@ -73,22 +73,23 @@ impl<Fp: Field> Constraints for KeccakEnv<Fp> {
 
         // SPONGE CONSTRAINTS
         {
-            for z in self.sponge_zeros() {
+            for i in 0..SPONGE_ZEROS_LEN {
                 // Absorb phase pads with zeros the new state
-                self.constrain(self.is_absorb() * z.clone());
+                self.constrain(self.is_absorb() * self.sponge_zeros()[i].clone());
```

Comment:
If this is going to be immediately cloned anyway, this makes sense to be a vector.